source/dol/bases/d_CourseSelectGuide.cpp

```diff
@@ -0,0 +1,1000 @@
+    };
+
+    static const char *GROUP_NAME_DT[ANIM_COUNT] = {
+        "A00_underGuide", "A01_zanki", "A02_courseInfo",
+        "A00_underGuide", "A01_zanki", "A02_courseInfo",
+        "A00_underGuide", "A01_zanki", "A02_courseInfo",
+        "B00_ArrowRight", "B01_ArrowLeft", "B02_ArrowUp", "B03_ArrowUnder",
+        "B00_ArrowRight", "B01_ArrowLeft", "B02_ArrowUp", "B03_ArrowUnder",
+        "B00_ArrowRight", "B01_ArrowLeft", "B02_ArrowUp", "B03_ArrowUnder",
+        "C00_shadow",
+        "C00_shadow"
+    };
+
+    static const int ANIME_INDEX_TBL[ANIM_COUNT] = {
+        inInfo, inInfo, inInfo,
+        loopInfo, loopInfo, loopInfo,
+        outInfo, outInfo, outInfo,
+        inMapArrow, inMapArrow, inMapArrow, inMapArrow,
+        loopArrow, loopArrow, loopArrow, loopArrow,
+        outMapArrow, outMapArrow, outMapArrow, outMapArrow,
+        inShadow,
+        outShadow
+    };
+
+    static const char *NPANE_NAME_DT[N_COUNT] = {
+        "N_IconPos1P_00", "N_IconPos2P_00", "N_IconPos3P_00", "N_IconPos4P_00",
+        "N_mapArrow_00",
+        "N_proportionL_00", "N_proportionR_00", "N_proportionC_00",
+        "N_guideViewC_00", "N_guideViewR_01",
+        "N_left_00"
+    };
+
+    static const char *T_PANE_NAME_TBL[T_COUNT] = {
+        "T_worldNum_00",
+        "T_cSelect_00", "T_cSelect_pic",
+        "T_lifeNumber_00", "T_lifeNumber_01", "T_lifeNumber_02", "T_lifeNumber_03",
+        "T_guideViewLS_00", "T_guideViewL_01"
+    };
+
+    static const int MESSAGE_DATA_TBL[] = {
+        MSG_CS_SELECT_WORLD,
+        MSG_CS_MENU,
+        MSG_CS_WORLD
+    };
+    static const char *T_PANE_FIXED_NAME_TBL[] = {
+        "T_guideViewC_00",
+        "T_guideViewR_00",
+        "T_world_00"
+    };
+
+    static const char *P_PANE_NAME_TBL[P_COUNT] = {
+        "P_cC_1_00",
+        "P_cC_2_00",
+        "P_cC_3_00",
+        "P_cC_1s_00",
+        "P_cC_2s_00",
+        "P_cC_3s_00",
+        "P_flagSkull_00",
+        "P_marioFace_00",
+        "P_luigiFace_00",
+        "P_BkinoFace_00",
+        "P_YkinoFace_00",
+        "P_bgShadow_00"
+    };
+
+    if (mInitialized) {
+        return true;
+    }
+    if (!mLayout.ReadResource("corseSelectUIGuide/corseSelectUIGuide.arc", false)) {
+        return false;
+    }
+    mLayout.build("corseSelectUIGuide_37.brlyt", 0);
+    mLayout.AnimeResRegister(AnmNameTbl, ARRAY_SIZE(AnmNameTbl));
+    mLayout.GroupRegister(GROUP_NAME_DT, ANIME_INDEX_TBL, ARRAY_SIZE(ANIME_INDEX_TBL));
+    mpRootPane = mLayout.getRootPane();
+    mLayout.NPaneRegister(NPANE_NAME_DT, &mpNullPanes[N_IconPos1P_00], ARRAY_SIZE(NPANE_NAME_DT));
+    mLayout.TPaneRegister(T_PANE_NAME_TBL, &mpTextBoxes[T_worldNum_00], ARRAY_SIZE(T_PANE_NAME_TBL));
+    mLayout.TPaneNameRegister(T_PANE_FIXED_NAME_TBL, MESSAGE_DATA_TBL, BMG_CATEGORY_COURSE_SELECT_GUIDE, ARRAY_SIZE(MESSAGE_DATA_TBL));
+    mLayout.PPaneRegister(P_PANE_NAME_TBL, &mpPicturePanes[P_cC_1_00], ARRAY_SIZE(P_PANE_NAME_TBL));
```

Comment:
Use the X_COUNT variables instead of the ARRAY_SIZE macro to reduce the line length and match other classes.